src/algebra/group_power/basic.lean

```diff
@@ -432,11 +440,18 @@ alias sub_sq ← sub_pow_two
 lemma sub_sq' (a b : R) : (a - b) ^ 2 = a ^ 2 + b ^ 2 - 2 * a * b :=
 by rw [sub_eq_add_neg, add_sq', neg_sq, mul_neg, ← sub_eq_add_neg]
 
+variables [no_zero_divisors R] {a b : R}
+
+lemma sq_eq_sq_iff_eq_or_eq_neg : a ^ 2 = b ^ 2 ↔ a = b ∨ a = -b :=
+(commute.all a b).sq_eq_sq_iff_eq_or_eq_neg
+
+lemma eq_or_eq_neg_of_sq_eq_sq (a b : R) : a ^ 2 = b ^ 2 → a = b ∨ a = -b :=
+sq_eq_sq_iff_eq_or_eq_neg.1
+
 /- Copies of the above comm_ring lemmas for `units R`. -/
 namespace units
 
-lemma eq_or_eq_neg_of_sq_eq_sq [no_zero_divisors R] (a b : Rˣ) (h : a ^ 2 = b ^ 2) :
-  a = b ∨ a = -b :=
+lemma eq_or_eq_neg_of_sq_eq_sq (a b : Rˣ) (h : a ^ 2 = b ^ 2) : a = b ∨ a = -b :=
```

Comment:
Can you add a `units.sq_eq_sq_iff_eq_or_eq_neg` too?

Reply:
Note that there were already some missing before, and that the "Copies" really was a singular one. There's only one lemma in this section and I don't really understand the point of having it. If we follow it, that means we should copy over every single ring lemma that doesn't mention addition/subtraction and zero. Do we want to do that?

Reply:
We used to have a lot more here, they were removed by `has_distrib_neg`

Reply:
> that means we should copy over every single ring lemma that doesn't mention addition/subtraction and zero. Do we want to do that?

So my answer would be "let's at least do that for new code", because if we end up with lots of lemmas here again that probably means it's time for another generalization

Reply:
That makes sense.